### ISSUE TITLE
Giant Project & VS Code Restart

### ISSUE THREAD

**Issue by bleshik** [fwcd/kotlin-language-server#160]
My current setup:
0) MacOS Mojave
1) VS Code Insiders
2) Remote SSH Extension
3) On the Remote Host the Kotlin extension is installed (which obviously uses kotlin-language-serve)

Problem:
The setup itself works, but not very usable. For a project Im currently working on it takes almost 2 hours to resolve dependencies before the server initialised. When the init process is done, the extension works just fine. However, the 2 hours init process is performed every time I connect to the remote host (or restart the vs code).

Expected behaviour:
After a full successful initialisation I'd expect the server to use the resolved dependencies event after the restart, the dependencies update may be done in background, while the server uses the (cached?) result after the previous initialisation. If there is such a caching ability, it would be very useful to have the description of how to set it up on README.

**Comment by fwcd** [comment]
Yeah, caching the dependencies seems like a reasonable solution to this. Do you use Gradle or Maven? In case you use Gradle, a daemon should keep running in the background providing fast, incremental build tasks.

**Comment by bleshik** [comment]
@fwcd I'm using Gradle, I think I've explicitly disabled daemons due to some issues, I'll give it a try with daemons a bit later. Thank you for the hint!
